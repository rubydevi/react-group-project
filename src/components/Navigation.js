import { NavLink } from 'react-router-dom';
import logo from '../images/planet.png';

const Navigation = () => (
  <div>
    <ul className="menu">
      <li className="logo">
        <img src={logo} alt="logo" />
        <h3>Space Travelers&apos; Hub</h3>
      </li>
      <li>
        <ul className="nav">
          <li>Rockets</li>
          <li>Missions</li>
          <li>Dragons</li>
          <li className="divider">|</li>
          <li>
            <NavLink className="nav-item" to="/profile" activeclassname="active">My Profile</NavLink>
          </li>
        </ul>
      </li>

    </ul>
  </div>
);

export default Navigation;
