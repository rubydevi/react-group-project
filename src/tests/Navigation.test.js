import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '../components/Navigation';

test('Navigation renders correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
