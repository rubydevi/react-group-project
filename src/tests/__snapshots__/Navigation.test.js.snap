// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div
  className="container"
>
  <ul
    className="menu"
  >
    <li
      className="logo"
    >
      <img
        alt="logo"
        src="planet.png"
      />
      <h3
        className="logo-text"
      >
        Space Travelers' Hub
      </h3>
    </li>
    <li>
      <ul
        className="nav"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            className=""
            href="/missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li>
          <a
            className=""
            href="/dragons"
            onClick={[Function]}
          >
            Dragons
          </a>
        </li>
        <span
          className="divider"
        >
          |
        </span>
        <li>
          <a
            className=""
            href="/profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
